v-next/hardhat-build-system/src/internal/utils/download.ts

```diff
@@ -19,48 +21,27 @@ function resolveTempFileName(filePath: string): string {
   });
 }
 
-export async function download(
-  url: string,
-  filePath: string,
-  timeoutMillis = 10000,
-  extraHeaders: { [name: string]: string } = {},
-) {
-  const { getGlobalDispatcher, ProxyAgent, request } = await import("undici");
-
-  let dispatcher: Dispatcher;
-  if (process.env.http_proxy !== undefined && shouldUseProxy(url)) {
-    dispatcher = new ProxyAgent(process.env.http_proxy);
-  } else {
-    dispatcher = getGlobalDispatcher();
+export async function download(url: string, filePath: string) {
```

Comment:
@kanej @alcuadrado @schaable check this function with caution, it downloads the compiler. Tests are passing, but it's worth spend some time on it